internal/reader/rmn_home.go

```diff
@@ -0,0 +1,331 @@
+type RmnHomePoller struct {
+	wg                   sync.WaitGroup
+	stopCh               services.StopChan
+	sync                 services.StateMachine
+	mutex                *sync.RWMutex
+	contractReader       contractreader.ContractReaderFacade
+	rmnHomeBoundContract types.BoundContract
+	lggr                 logger.Logger
+	rmnHomeState         rmnHomeState
+	failedPolls          uint
+	pollingDuration      time.Duration // How frequently the poller fetches the chain configs
+}
+
+func NewRMNHomePoller(
+	contractReader contractreader.ContractReaderFacade,
+	rmnHomeBoundContract types.BoundContract,
+	lggr logger.Logger,
+	pollingInterval time.Duration,
+) RMNHome {
+	return &RmnHomePoller{
+		stopCh:               make(chan struct{}),
+		contractReader:       contractReader,
+		rmnHomeBoundContract: rmnHomeBoundContract,
+		rmnHomeState:         rmnHomeState{},
+		mutex:                &sync.RWMutex{},
+		failedPolls:          0,
+		lggr:                 lggr,
+		pollingDuration:      pollingInterval,
+	}
+}
+
+func (r *RmnHomePoller) Start(ctx context.Context) error {
+	r.lggr.Infow("Start Polling RMNHome")
+	return r.sync.StartOnce(r.Name(), func() error {
+		r.wg.Add(1)
+		go r.poll()
+		return nil
+	})
+}
+
+func (r *RmnHomePoller) poll() {
+	defer r.wg.Done()
+	ctx, cancel := r.stopCh.NewCtx()
+	defer cancel()
+	// Initial fetch once poll is called before any ticks
+	if err := r.fetchAndSetRmnHomeConfigs(ctx); err != nil {
+		// Just log, don't return error as we want to keep polling
+		r.lggr.Errorw("Initial fetch of on-chain configs failed", "err", err)
+	}
+
+	ticker := time.NewTicker(r.pollingDuration)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			r.mutex.Lock()
+			r.failedPolls = 0
+			r.mutex.Unlock()
+			return
+		case <-ticker.C:
+			if err := r.fetchAndSetRmnHomeConfigs(ctx); err != nil {
+				r.mutex.Lock()
+				r.failedPolls++
+				r.mutex.Unlock()
+			}
+		}
+	}
+}
+
+func (r *RmnHomePoller) fetchAndSetRmnHomeConfigs(ctx context.Context) error {
+	var versionedConfigWithDigests []rmntypes.VersionedConfigWithDigest
+	err := r.contractReader.GetLatestValue(
+		ctx,
+		r.rmnHomeBoundContract.ReadIdentifier(consts.MethodNameGetAllConfigs),
+		primitives.Unconfirmed,
+		map[string]interface{}{},
+		&versionedConfigWithDigests,
+	)
+	if err != nil {
+		return fmt.Errorf("error fetching RMNHomeConfig: %w", err)
+	}
+
+	if len(versionedConfigWithDigests) != expectedRMNHomeConfigs {
+		r.lggr.Warnw("expected 2 RMNHomeConfigs, got", "count", len(versionedConfigWithDigests))
+		return fmt.Errorf("expected 2 RMNHomeConfigs, got %d", len(versionedConfigWithDigests))
+	}
+
+	primaryConfigDigest := versionedConfigWithDigests[0].ConfigDigest
+	if primaryConfigDigest.IsEmpty() {
+		r.lggr.Warnw("primary config digest is empty")
+	}
+
+	secondaryConfigDigest := versionedConfigWithDigests[1].ConfigDigest
+	if secondaryConfigDigest.IsEmpty() {
+		r.lggr.Warnw("secondary config digest is empty")
```

Comment:
I don't know exactly how this works, but would be nice to not have `warn` logs when everything seems ok. Are we always expecting to have a secondary config digest?

Reply:
Just a disclaimer, contracts are not ready. But from the what I can tell, primary is acting as blue and secondary as green. So in most of the cases, we are going to have only 1 config set. But during migration 2 are going to set until we check that everything is working properly and then we are going to remove the previous config.

Reply:
Yes we can have a debug log in this case.